src/redux/categories/Category.js

```diff
@@ -1,28 +1,51 @@
 import React from 'react';
+import { createSelector } from '@reduxjs/toolkit';
 import { useSelector } from 'react-redux';
 import { useParams, Link } from 'react-router-dom';
 import { selectCategoryById } from './categoriesSlice';
-import { selectBooksByCategory } from '../books/booksSlice';
+import { selectAllBooks } from '../books/booksSlice';
 
 function Category() {
+  /*  const { categoryId } = useParams();
+  console.log(categoryId);
+  const category = useSelector((state) => selectCategoryById(state, Number(categoryId)));
+  console.log(category);
+
+  const selectBooksByCategory = createSelector(
+    [selectAllBooks, (_, categoryId) => categoryId],
+    (books, categoryId) => {
+      const booksbyCategory = Object.keys(books)
+        .filter((bookId) => books[bookId].category === categoryId)
+        .reduce((filteredBooks, bookId) => {
+          // eslint-disable-next-line no-param-reassign
+          filteredBooks[bookId] = { ...books[bookId] };
+          return filteredBooks;
+        }, {});
+      return booksbyCategory;
+    },
+  );
```

Comment:
### Optional
- [ ] It would be nice if we remove all commented piece of code, it makes our codebase looks messy and difficult to understand